pkg/cmd/grafana-cli/commands/install_command.go

```diff
@@ -105,93 +129,102 @@ func InstallPlugin(pluginName, version string, c utils.CommandLine) error {
 	return err
 }
 
-func SelectVersion(plugin m.Plugin, version string) (m.Version, error) {
+func osAndArchString() string {
+	osString := strings.ToLower(runtime.GOOS)
+	arch := runtime.GOARCH
+	return osString + "-" + arch
+}
+
+func supportsCurrentArch(version *m.Version) bool {
+	if version.Arch == nil {
+		return true
+	}
+	for arch := range version.Arch {
+		if arch == osAndArchString() || arch == "any" {
+			return true
+		}
+	}
+	return false
+}
+
+func latestSupportedVersion(plugin *m.Plugin) *m.Version {
+	for _, ver := range plugin.Versions {
+		if supportsCurrentArch(&ver) {
+			return &ver
+		}
+	}
+	return nil
+}
+
+// SelectVersion returns latest version if none is specified or the specified version. If the version string is not
+// matched to existing version it errors out. It also errors out if version that is matched is not available for current
+// os and platform.
+func SelectVersion(plugin *m.Plugin, version string) (*m.Version, error) {
+	var ver *m.Version
 	if version == "" {
-		return plugin.Versions[0], nil
+		ver = &plugin.Versions[0]
 	}
 
 	for _, v := range plugin.Versions {
 		if v.Version == version {
-			return v, nil
+			ver = &v
 		}
 	}
 
-	return m.Version{}, errors.New("Could not find the version you're looking for")
+	if ver == nil {
+		return nil, xerrors.New("Could not find the version you're looking for")
+	}
+
+	latestForArch := latestSupportedVersion(plugin)
+	if latestForArch.Version == ver.Version {
+		return ver, nil
+	}
+	return nil, xerrors.Errorf("Version you want is not supported on your architecture. Latest suitable version is %v", latestForArch.Version)
```

Comment:
In the future this should be changed to `errors` and I think it should no be much work to just do `s/xerrors/errors` globally. Aliasing everything behind errutils whether transiently or forever does not seem worth it to me, either we will need to rename it later too or we have alias for otherwise standard go package which can be confusing.